### PR TITLE
Use unitTurret command

### PR DESCRIPTION
**Pull request by PabstMirror**

### CODE REVIEW

File: addons/events/fnc_addPlayerEventHandler.sqf [hunk 1]
```diff
@@ -101,10 +98,7 @@ private _id = switch (_type) do {
     case "turret": {
         if (_applyRetroactively) then {
             private _vehicle = vehicle GVAR(oldUnit);
-            private _turret = [];
-            if (GVAR(oldUnit) != _vehicle) then {
-                _turret = ([[-1]] + allTurrets [_vehicle, true]) select {_vehicle turretUnit _x == GVAR(oldUnit)} param [0, []];
-            };
+            private _turret = _vehicle unitTurret GVAR(oldUnit);
```
Comment (jonpas) [inline addons/events/fnc_addPlayerEventHandler.sqf:101]:
```suggestion
            private _turret = _vehicle turretUnit GVAR(oldUnit);
```

File: addons/events/fnc_addPlayerEventHandler.sqf [hunk 2]
```diff
@@ -66,10 +66,7 @@ private _id = switch (_type) do {
     case "turretweapon": {
         if (_applyRetroactively) then {
             private _vehicle = vehicle GVAR(oldUnit);
-            private _turret = [];
-            if (GVAR(oldUnit) != _vehicle) then {
-                _turret = ([[-1]] + allTurrets [_vehicle, true]) select {_vehicle turretUnit _x == GVAR(oldUnit)} param [0, []];
-            };
+            private _turret = _vehicle unitTurret GVAR(oldUnit);
```
Comment (jonpas) [inline addons/events/fnc_addPlayerEventHandler.sqf:69]:
```suggestion
            private _turret = _vehicle turretUnit GVAR(oldUnit);
```

File: addons/events/fnc_playerEvent.sqf [hunk 3]
```diff
@@ -37,11 +37,8 @@ if (!isNull getConnectedUAV _unit) then {
     };
 };
 
-private _turret = [];
-if (_unit != _vehicle) then {
-    // Unlike CBA_fnc_turretPath, this will return [-1] when player is driver
-    _turret = ([[-1]] + allTurrets [_vehicle, true]) select {_vehicle turretUnit _x == _unit} param [0, []];
-};
+// Unlike CBA_fnc_turretPath, this will return [-1] when player is driver
+private _turret = _vehicle unitTurret _unit;
```
Comment (jonpas) [inline addons/events/fnc_playerEvent.sqf:41]:
```suggestion
private _turret = _vehicle turretUnit _unit;
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/events/fnc_addPlayerEventHandler.sqf (hunk 1 above)
Comment (PabstMirror) [reply addons/events/fnc_addPlayerEventHandler.sqf:101]: I get confused by these names (even though I guess it makes sense) but what is there now should be right
- unitTurret = turret path (array) [the new command]
- turretUnit = unit in turret (object)

Comment (jonpas) [reply addons/events/fnc_addPlayerEventHandler.sqf:101]: Ah right! :+1: I just thought your last commit broke them.

**Review by commy2 — APPROVED** [review]

File: addons/common/fnc_turretPath.sqf [hunk 4]
```diff
@@ -27,4 +27,9 @@ params [["_unit", objNull, [objNull]]];
 
 private _vehicle = vehicle _unit;
 
-(allTurrets [_vehicle, true] select {(_vehicle turretUnit _x) isEqualTo _unit}) param [0, []]
+private _turret = _vehicle unitTurret _unit;
+
+// driver will be returned as [] to be consistant with previous behaviour
```
Comment (CreepPork) [inline addons/common/fnc_turretPath.sqf:32]:
```suggestion
// driver will be returned as [] to be consistent with previous behaviour
```
There's a small typo.

**Review by jonpas — APPROVED** [review]

**Pull request merged**